### PR TITLE
add bounties

### LINKED ISSUE #199: enable bounties to activate developer ecosystem
Deploy the bounties pallet and configure it similar to Kusama, including child bounties because Integritee AG will use child-bounties to spin out small tasks directly (itself curating) to the developer ecosystem. 

The role of integritee council is to define the amount in the pot available for bounties for a given time > 3mt

The role of the curator (employee(s) of Integritee AG) is to identify good isolated issues to be solved by devs outside the core team and select solvers, answer questions, review and merge PR's

intended process (non-restrictive because unpermissioned):
1. Integritee AG (or s/o else) suggests a bounty pot `bounties:propose_bounty` to compensate the dev community for work on individually assigned github tasks. Integritee will assign a pure proxy as curator which delegates control to 2-3 employees of Integritee AG
2. The council of Integritee network will decide on bounty proposals (`MoreThanHalfCouncil`)
3. The curator will create new child bounties:
   1. label github issues with `T0-apply-for-bounty` 
   2. child bounty submitted with `child_bounties:add_child_bounty` 
   3. (self-)assigns a child curator
   4. link GH issue on Polkassembly
   5. link Polkassembly child bounty page in comment on GH
4. multiple solvers can apply for the bounty by commenting on the issue and declaring their interest and qualification
5. the child bounty curator will assign an issue to one solver only and change the label to `T1-bounty-in-process`
6. the assigned solver submits a PR that resolves the issue
7. the curator reviews the PR
8. upon merge of the PR, the child-bounty is paid out to the solver
9. upon confirmation of payout, curator changes label to `T2-bounty-paid` on the (now closed) issue

# TODO
* instantiate bounties and child-bounties pallets and configure origins to support the above scenario
* test on local instance
* deploy and test with polkassembly flow using a trivial issue leading to a real PR which will be resolved internally

### PR DESCRIPTION
**Pull request by brenzi**
closes #199 

Test protocol: https://integritee.atlassian.net/wiki/spaces/INTEGRITEE/pages/516947992/Test+Treasury+and+Bounties

tl;dr
Rococo-local setup works fine. bounties and child bounties tested until completion and payout. However, Child Bounties are not supported by js/apps and Events by that pallet are a bit minimalistic, so testing is low-level involved. Luckily, they are supported on Polkassembly, but this we can only test on mainnet

### CODE REVIEW

File: polkadot-parachains/integritee-runtime/src/lib.rs [hunk 1]
```diff
@@ -523,10 +532,48 @@ impl pallet_treasury::Config for Runtime {
...
 	type MaxApprovals = MaxApprovals; //0:cannot approve any proposal
 	type WeightInfo = weights::pallet_treasury::WeightInfo<Runtime>;
-	type SpendOrigin = frame_support::traits::NeverEnsureOrigin<Balance>; // Same as kusama
+}
+
+parameter_types! {
+	pub const BountyDepositBase: Balance = 1 * TEER;
+	pub const BountyDepositPayoutDelay: BlockNumber = prod_or_fast!(4 * DAYS, 4 * MINUTES);
+	pub const BountyUpdatePeriod: BlockNumber = prod_or_fast!(90 * DAYS, 15 * MINUTES);
+	pub const MaximumReasonLength: u32 = 16384;
```
Comment (coax1d) [inline polkadot-parachains/integritee-runtime/src/lib.rs:545]: Is 16kb enough for bounty proposals? Is this in our case just intended to be the encoded text for the proposal or something else? I suppose some upper bound needs to be chosen and if it is just characters then 16kb probably is sufficient.

**Review by coax1d — COMMENTED** [review]
Everything looks straightforward and good. Just a quick question/note on the configuration of the bounties reason field.

File: polkadot-parachains/integritee-runtime/src/lib.rs (hunk 1 above)
Comment (brenzi) [reply polkadot-parachains/integritee-runtime/src/lib.rs:545]: actually, I think it should be enough to carry a url of a github issue. but not more than that and certainly not an entire description, that would just be chain bloat, so I guess the current length is plenty. 
As people pay fees per length, I think it should be safe as is.

Comment (coax1d) [reply polkadot-parachains/integritee-runtime/src/lib.rs:545]: Ok a link seems reasonable yea I agree anything more is probably worthy of a hash instead of just a bunch of description bytes. 

I did not think it was dangerous by any means just was more of a curiosity.

**Review by coax1d — APPROVED** [review]

File: polkadot-parachains/integritee-runtime/Cargo.toml [hunk 2]
```diff
@@ -37,6 +37,8 @@ frame-system-rpc-runtime-api = { default-features = false, git = "https://github
 # pallets
 pallet-aura = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
 pallet-balances = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+pallet-bounties = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+pallet-child-bounties = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
 pallet-collective = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
```
Comment (clangenb) [inline polkadot-parachains/integritee-runtime/Cargo.toml:42]: std enablement missing

Comment (coax1d) [reply polkadot-parachains/integritee-runtime/Cargo.toml:42]: Had no idea the CI would pass without the std checks. Good thing this was pointed out for the future

Comment (brenzi) [reply polkadot-parachains/integritee-runtime/Cargo.toml:42]: that's why CI failed I guess
